active/teachingtip/TeachingTipSpec.md

```diff
@@ -0,0 +1,565 @@
+            Title="Saving automatically"
+            Subtitle="We save your changes as you go - so you never have to."
+            PreferredPlacement="BottomEdgeAlignedLeft">
+        </muxc:TeachingTip>
+    </Button.Resources>
+</Button>
+```
+
+![A sample app with a teaching tip pointing upward at a button that reads "Save". The left edge of the teaching tip and the "Save" button are aligned. The tip title reads "Saving automatically" and the subtitle reads "We save your changes as you go - so you never have to." There is a close button on the top right corner of the teaching tip.](TeachingTipPointingPreferredPlacementSample.jpg)
+
+
+A non-pointing teaching tip with its PreferredPlacement set to "BottomEdgeAlignedLeft" will appear in the bottom left corner of the XamlRoot.
+
+XAML
+```XAML
+<Button x:Name="SaveButton" Content="Save" />
+
+<muxc:TeachingTip x:Name="AutoSaveTip"
+    Title="Saving automatically"
+    Subtitle="We save your changes as you go - so you never have to."
+    PreferredPlacement="BottomEdgeAlignedLeft">
+</muxc:TeachingTip>
+```
+
+![A sample app with a teaching tip in the bottom left corner. The tip title reads "Saving automatically" and the subtitle reads "We save your changes as you go - so you never have to." There is a close button on the top right corner of the teaching tip.](TeachingTipNonPointingPreferredPlacementSample.jpg)
+
+### Escaping XamlRoot bounds
+
+On Windows version 19H1 and above, a teaching tip can escape XamlRoot bounds by setting the ShouldConstrainToRootBounds property. On earlier versions of Windows, this property is ignored and the teaching tip always stays within the XamlRoot bounds.
+
+XAML
+```XAML
+<Button x:Name="SaveButton" Content="Save" />
+
+<muxc:TeachingTip x:Name="AutoSaveTip"
+    Title="Saving automatically"
+    Subtitle="We save your changes as you go - so you never have to."
+    PreferredPlacement="BottomEdgeAlignedRight"
+    TargetOffset="-80,-50,0,0"
+    ShouldConstrainToRootBounds="False">
+</muxc:TeachingTip>
+```
+
+![A sample app with a teaching tip outside of the app's bottom right corner. The tip title reads "Saving automatically" and the subtitle reads "We save your changes as you go - so you never have to." There is a close button on the top right corner of the teaching tip.](TeachingTipOutOfWindowBoundsSample.jpg)
+
+### Canceling and deferring close
+
+The Closing event can be used to cancel and/or defer the close of a teaching tip. This can be used to keep the teaching tip open or allow time for an action or custom animation to occur. 
+
+XAML
+```XAML
+<muxc:TeachingTip x:Name="EnableNewSettingsTip"
+    Title="New ways to protect your privacy!"
+    Subtitle="Please close this tip and review our updated privacy policy and privacy settings."
+    Closing="OnTipClosing">
+</muxc:TeachingTip>
+```
+
+C#
+```C#
+public void OnTipClosing(object sender, TeachingTipClosingEventArgs args)
+{
+    if (args.Reason == TeachingTipCloseReason.CloseButton)
```

Comment:
Can a Programmatic close be canceled?

Reply:
Added the explanation [here](https://github.com/Microsoft/microsoft-ui-xaml-specs/commit/52887dcce86fc52198705b34d3b6e137a4688ee9).